### ISSUE TITLE
Implications of warnings:

### ISSUE THREAD

**Issue by csverma610** [libigl/libigl#284]
Hello,

When we compile libIGL with Eigen3.2.8, we get a large number of warnings similar to the following.
Are there serious implications of such warnings ? 

/home/csverma/Software/Mesh/LibIGL/include/igl/cat.cpp:41:41: warning: ‘Eigen::DynamicSparseMatrix<_Scalar, _Flags, _Index>::DynamicSparseMatrix() [with _Scalar = double; int _Options = 1; _Index = int]’ is deprecated [-Wdeprecated-declarations]
   DynamicSparseMatrix<Scalar, RowMajor> dyn_C;

Thanks.
csv

**Comment by alecjacobson** [comment]
In cat.cpp this should be gone, as of
56bdf2c2a4a6aab2cfe7e3c3c2b55f7a6d8b4516.

These are deprecation warnings because (I believe) Eigen is planning to
phase out the DynamicSparseMatrix class. We kept some implementations in
libigl that used it because it was faster than the default use of
SparseMatrix. I have replaced DynamicSparseMatrix in igl::cat and
igl::repdiag recently, but it's still used in igl::slice and
igl::slice_into for performance reasons.

So bottom line: for Eigen 2.8 this can be ignored, for some future Eigen
version DynamicSparseMatrix may disappear and these functions will break.
